engine/src/main/resources/org/camunda/bpm/engine/impl/mapping/entity/HistoricTaskInstance.xml

```diff
@@ -738,6 +738,7 @@
       <if test="parameter.finishedAfter == null and parameter.finishedAt == null">
         and RES.END_TIME_ is not null
       </if>
+      and TASK_DEF_KEY_ is not null
```

Comment:
Q for reviwer:
The name of this file is pretty generic but I saw these queries are only used for the optimize service, which is why I hardcoded this exclusion. Am I correct in assuming thats what this file is for? If it could also be used for other scenarios I would go ahead and adjust this to not hardcode the exclusion of standalone tasks but instead make it a parameterised flag.